wrangler-service/src/main/java/co/cask/wrangler/service/bigquery/BigQueryHandler.java

```diff
@@ -180,41 +171,59 @@ public void listDatasets(HttpServiceRequest request, HttpServiceResponder respon
+    listTables(request, responder, getContext().getNamespace(), connectionId, datasetStr);
+  }
 
-    if (!validateConnection(connectionId, connection, responder)) {
-      return;
-    }
-    BigQuery bigQuery = GCPUtils.getBigQueryService(connection);
+  /**
+   * List all tables in a dataset.
+   *
+   * @param request HTTP requets handler.
+   * @param responder HTTP response handler.
+   * @param datasetStr the dataset id as a string. It will be of the form [project:]name.
+   *   The project prefix is optional. When not given, the connection project should be used.
+   */
+  @GET
+  @Path("contexts/{context}/connections/{connection-id}/bigquery/{dataset-id}/tables")
+  public void listTables(HttpServiceRequest request, HttpServiceResponder responder,
+                         @PathParam("context") String namespace,
+                         @PathParam("connection-id") String connectionId,
+                         @PathParam("dataset-id") String datasetStr) {
+    respond(request, responder, namespace, () -> {
+      Connection connection = store.get(connectionId);
+      validateConnection(connectionId, connection);
+      BigQuery bigQuery = GCPUtils.getBigQueryService(connection);
 
-    DatasetId datasetId = getDatasetId(datasetStr, GCPUtils.getProjectId(connection));
+      DatasetId datasetId = getDatasetId(datasetStr, GCPUtils.getProjectId(connection));
 
-    try {
-      Page<Table> tablePage = bigQuery.listTables(datasetId);
-      List<TableInfo> values = new ArrayList<>();
-      for (Table table : tablePage.iterateAll()) {
-        values.add(new TableInfo(table.getTableId().getTable(), table.getFriendlyName(), table.getDescription(),
-                                 table.getEtag(), table.getCreationTime(), table.getLastModifiedTime(),
-                                 table.getExpirationTime()));
-      }
+      try {
+        Page<Table> tablePage = bigQuery.listTables(datasetId);
+        List<TableInfo> values = new ArrayList<>();
+        for (Table table : tablePage.iterateAll()) {
+          values.add(new TableInfo(table.getTableId().getTable(), table.getFriendlyName(), table.getDescription(),
+                                   table.getEtag(), table.getCreationTime(), table.getLastModifiedTime(),
+                                   table.getExpirationTime()));
+        }
 
-      ServiceResponse<TableInfo> response = new ServiceResponse<>(values);
-      responder.sendJson(response);
-    } catch (BigQueryException e) {
-      if (e.getReason() != null) {
-        // CDAP-14155 - BigQueryException message is too large. Instead just throw reason of the exception
-        throw new RuntimeException(e.getReason());
+        return new ServiceResponse<>(values);
+      } catch (BigQueryException e) {
+        if (e.getReason() != null) {
+          // CDAP-14155 - BigQueryException message is too large. Instead just throw reason of the exception
+          throw new RuntimeException(e.getReason());
```

Comment:
same question for line 354.

Reply:
reason is a string, so this mostly the same behavior as before, except that the response format will be consistent with other error scenarios as a json object instead of a string.